src/rdf4cpp/rdf/writer/NQuadsWriter.cpp

```diff
@@ -14,7 +14,7 @@ NQuadsWriter::operator std::string() const {
 }
 std::ostream &operator<<(std::ostream &os, const NQuadsWriter &writer) {
     for (const Quad &quad : writer.dataset_) {
-        if (not(quad.graph().is_iri() and (static_cast<IRI>(quad.graph())) == IRI::default_graph(writer.dataset_.backend().node_storage())))
+        if (!quad.graph().null() && quad.graph().as_iri() != IRI::default_graph(writer.dataset_.backend().node_storage()))
```

Comment:
```suggestion
        if (quad.graph().as_iri() != IRI::default_graph(writer.dataset_.backend().node_storage()))
```

Should be sufficient, right?

Reply:
but `null-iri != default_graph_iri` and then we would try to print a null-iri

Reply:
you are right